### PR TITLE
Fix spelling and grammar

### PR DESCRIPTION
**Pull request by bmesuere**
This pull request cleans up the English translation strings of Dodona.

The initial goal was to only fix https://github.com/dodona-edu/dodona/discussions/4240#discussioncomment-4397438 but I found several other issues.

### CODE REVIEW

File: config/locales/views/judges/en.yml [hunk 1]
```diff
@@ -1,8 +1,8 @@
 en:
   judges:
     form:
-      remote_help: "The SSH clone URL of the judge repository. Dodona needs access rights to the repository to be able to clone it."
-      image_help: "The name of the Docker image in which the judge will be started. The image will be fetched from docker hub if it isn't available locally."
+      remote_help: "The SSH clone URL of the judge repository. Dodona needs access rights to the repository in order to clone it."
+      image_help: "The name of the Docker image in which the judge will be started. The image will be fetched from Docker Hub if it isn't available locally."
```
Comment (chvp) [inline config/locales/views/judges/en.yml:5]: Not really related to this PR, but this is not true.

File: config/locales/views/defaults/en.yml [hunk 2]
```diff
@@ -17,16 +17,16 @@ en:
...
-      info:  while in anonymous mode, Dodona conceals all users names.
+      info:  while in anonymous mode, Dodona conceals the names of all users.
       switch_back: Disable anonymous mode.
     messages:
       warning: Warning!
       danger: Oh snap!
       info: "Heads up:"
       success: Great!
-      time_zone_html: Your time zone is currently set to <strong>%{time_zone}</strong>, but this doesn't seem to match with your local time. You can switch time zones on your <a href='%{profile_edit}'>profile page</a>.
+      time_zone_html: Your timezone is currently set to <strong>%{time_zone}</strong>, but this doesn't seem to match with your local time. You can change your time zone on your <a href='%{profile_edit}'>profile page</a>.
```
Comment (niknetniko) [inline config/locales/views/defaults/en.yml:27]:
```suggestion
      time_zone_html: Your timezone is currently set to <strong>%{time_zone}</strong>, but this doesn't seem to match with your local time. You can change your timezone on your <a href='%{profile_edit}'>profile page</a>.
```

**Review by niknetniko — APPROVED** [review]

File: config/locales/views/courses/en.yml [hunk 3]
```diff
@@ -9,17 +9,17 @@ en:
       questions: Questions from students
     form:
       teacher-help: Optional. Can help users find your course.
-      year-help: Using a year of the form yyyy-yyyy will make sure your course gets sorted correctly.
+      year-help: Using a year in the form yyyy-yyyy will ensure that your course is sorted correctly.
```
[the remark this answers is not in the document]
Comment (bmesuere) [reply config/locales/views/courses/en.yml:12]: I checked online and in seems more common

File: config/locales/views/judges/en.yml (hunk 1 above)
Comment (bmesuere) [inline config/locales/views/judges/en.yml:5]:
```suggestion
      image_help: "The name of the Docker image in which the judge will be started."
```

**Pull request merged**